### ISSUE TITLE
Error doing Cmake

### ISSUE THREAD

**Issue by Rovi98** [tensorflow/mlir#190]
When I execute:
`cmake -G Ninja ../llvm -DLLVM_BUILD_EXAMPLES=ON -DLLVM_TARGETS_TO_BUILD="host`

I  have the following error
```
-- Targeting X86
CMake Error at cmake/modules/LLVMProcessSources.cmake:112 (message):
  Found unknown source file AffineLoopInvariantCodeMotion.cpp

  Please update
  /home/rovi98/llvm-project/llvm/projects/mlir/lib/Transforms/CMakeLists.txt

Call Stack (most recent call first):
  cmake/modules/LLVMProcessSources.cmake:62 (llvm_check_source_file_list)
  cmake/modules/AddLLVM.cmake:415 (llvm_process_sources)
  cmake/modules/AddLLVM.cmake:683 (llvm_add_library)
  projects/mlir/lib/Transforms/CMakeLists.txt:3 (add_llvm_library)
```

**Comment by bondhugula** [comment]
The commit https://github.com/tensorflow/mlir/commit/a266aec1c35aca193df9ed2517d1c37aa1d895a5 appears to be missing all changes to the build rules. 

1) AffineLoopInvariantCodeMotion.cpp should be added to lib/Transforms/CMakeLists.txt
2) include/mlir/CMakeLists.txt, lib/Transforms/CMakeLists.txt need an update,  and a include/mlir/Transforms/CMakeLists.txt is needed to generate mlir/Transforms/LoopLikeInterface.h.inc. This changeset will make it build. 

```
diff --git a/include/mlir/CMakeLists.txt b/include/mlir/CMakeLists.txt
index ccaa7177..1a5094df 100644
--- a/include/mlir/CMakeLists.txt
+++ b/include/mlir/CMakeLists.txt
@@ -1,3 +1,4 @@
 add_subdirectory(Analysis)
 add_subdirectory(Dialect)
 add_subdirectory(EDSC)
+add_subdirectory(Transforms)
diff --git a/lib/Transforms/CMakeLists.txt b/lib/Transforms/CMakeLists.txt
index 418b5166..746c608a 100644
--- a/lib/Transforms/CMakeLists.txt
+++ b/lib/Transforms/CMakeLists.txt
@@ -2,6 +2,7 @@ add_subdirectory(Utils)
 
 add_llvm_library(MLIRTransforms
   AffineDataCopyGeneration.cpp
+  AffineLoopInvariantCodeMotion.cpp
   AffineScalarReplacement.cpp
   Canonicalizer.cpp
   CSE.cpp
@@ -28,7 +29,10 @@ add_llvm_library(MLIRTransforms
   ${MLIR_MAIN_INCLUDE_DIR}/mlir/Transforms
   )
 
-add_dependencies(MLIRTransforms MLIRStandardOpsIncGen)
+add_dependencies(MLIRTransforms
+  MLIRStandardOpsIncGen
+  MLIRLoopLikeInterfaceIncGen
+  )
 target_link_libraries(MLIRTransforms
   MLIRAffineOps
   MLIRAnalysis

diff --git a/include/mlir/Transforms/CMakeLists.txt b/include/mlir/Transforms/CMakeLists.txt
new file mode 100644
index 00000000..63fdf22f
--- /dev/null
+++ b/include/mlir/Transforms/CMakeLists.txt
@@ -0,0 +1,4 @@
+set(LLVM_TARGET_DEFINITIONS LoopLikeInterface.td)
+mlir_tablegen(LoopLikeInterface.h.inc -gen-op-interface-decls)
+mlir_tablegen(LoopLikeInterface.cpp.inc -gen-op-interface-defs)
+add_public_tablegen_target(MLIRLoopLikeInterfaceIncGen)
```

**Comment by joker-eph** [comment]
Fixed in 43af5a5f84e92b16ca128aaf178b7c053325d606

FYI you can see the green/red checkbox next to each commits here: https://github.com/tensorflow/mlir/commits/master

These indicates the CI status, and clicking on these gives more information about the state of the build. In general when it is red we are already working on a fix.

**Comment by Rovi98** [comment]
Thanks a lot 😊
